### PR TITLE
Update default layout

### PR DESCRIPTION
**Pull request by dreadnip**
This PR updates all shared parts of the layout (header, nav, footer) and includes a basic CSS reset.

I have more work prepared for each of the main pages (docs, screenshots, downloads), but they all kind of depend on these initial 3 PRs. Then, after minimal changes to the major subpages, I can send one more PR that includes media queries and then the site should be mobile ready.

### CODE REVIEW

File: css/style.css [hunk 1]
```diff
@@ -24,59 +36,55 @@ a {
...
+    padding: .4rem .8rem;
     text-decoration: none;
     border-bottom: 2px solid #2e2e2e;
+    transition: ease 150ms;
 }
 
-#nav li a:hover {
-    border-bottom: 2px solid #fff;
+nav ul li a:hover {
+    border-bottom: 2px solid #187b55;
```
Comment (stapelberg) [inline css/style.css]: Let’s stick with the white border-bottom here, to not introduce too many different colors

File: css/style.css [hunk 2]
```diff
@@ -1,9 +1,21 @@
...
+    margin: 0;
+    padding: 0;
+    box-sizing: border-box;
+}
+
 body {
     background-color: #2e2e2e;
     color: #CCC;
     font-family: 'Droid Sans', "Lucida Grande", "Lucida Sans", "Liberation Sans", sans-serif;
     font-size: 0.8em;
```
Comment (stapelberg) [inline css/style.css]: I don’t know extactly where to place this comment, but with this PR, the font is noticeably smaller (12px instead of 15px) than before. Can we keep the current font size?

Comment (dreadnip) [reply css/style.css]: The font is smaller because the base font-size of the body element on this branch is still set to 0.8em. The #content rule has a 1.2em font-size rule which ended up normalizing the font size to 1em, but with the removal of #content on the main pages you end up seeing 0.8em for now. Once the typography branch is merged, the base font-size will be 1em and things should look like they used to. #content will need some adjusting afterwards, like setting the font-size back to 1em and removing the redundant line-height, but I felt like that wasn't an appropriate change for this branch,

File: css/style.css (hunk 1 above)
Comment (dreadnip) [reply css/style.css]: Reverted in ecfc2b99468f4082082a68c67ad5581804715415

**Comment by dreadnip** [comment]
I tried to rebase from the upstream master branch once the typo branch got merged to solve any possible merge conflicts, but it seems like I messed up and now I see a bunch of duplicate commits. If I need to undo something let me know.

**Comment by stapelberg** [comment]
High level feedback from the first look:

1. On the main page, there isn’t enough spacing above the download button.

1. On the docs page, there also needs to be more spacing—everything looks pretty cramped.

1. E.g. /docs/userguide.html doesn’t render correctly (maybe a few required CSS classes were removed accidentally?)

**Comment by dreadnip** [comment]
For point 1 & 2: I've already fixed the problems with each specific page on a separate branch, which I was planning to PR after this general one. If you want I can cherry-pick the commits, or just port the code over to this one. The bulk of the changes is removing floats and using flexbox for the layout of the homepage infobox, downloads page, screenshots page and docs overview page.

3: I haven't touched the docs because I saw that their HTML files are generated and don't use the default Jekyll template. 

From what I have seen we have 2 options here to modernize the HTML code behind the docs; either figure out a way to generate the doc files as Jekyll templates, so with:
```
---
layout: default
title: Docs - Userguide
---
```
at the top. This seems pretty unlikely, as I don't know how flexible the asciidoc CLI is, but maybe you have more experience with this.

The second option would be to update the asciidoc back-end from xhtml1 to html5, although from what I can see on [this page](http://methods.co.nz/asciidoc/userguide.html#X5), the html5 back-end is 'functionally identical' to the xhtml1 one, so I don't know how much of an upgrade that would be. I looked further and bumped into [asciidoctor](https://asciidoctor.org/), which seems more modern and is also what Github uses to support AsciiDoc format. Are you ok with me looking into this a bit further and finding a way to output cleaner HTML for the docs?

For backwards compatibility we could keep the xhtml1.css file and keep it linked to all the docs that are already built, and link to the new style.css file for any newer docs.

**Comment by stapelberg** [comment]
> For point 1 & 2: I've already fixed the problems with each specific page on a separate branch, which I was planning to PR after this general one. If you want I can cherry-pick the commits, or just port the code over to this one. The bulk of the changes is removing floats and using flexbox for the layout of the homepage infobox, downloads page, screenshots page and docs overview page.

Well, any PR I merge will be live within seconds. Hence, let’s do the changes in such a way that they never break something please :)

> 3: I haven't touched the docs because I saw that their HTML files are generated and don't use the default Jekyll template.
> 
> From what I have seen we have 2 options here to modernize the HTML code behind the docs; either figure out a way to generate the doc files as Jekyll templates, so with:
> 
> ```
> ---
> layout: default
> title: Docs - Userguide
> ---
> ```
> 
> at the top. What if we place this snippet in the [header] tag of conf/i3html.conf. Would that work? I'm not that familiar with the asciidoc CLI so I don't know how flexible it is.

I don’t know. Only one way to find out :)

> The second option would be to update the asciidoc back-end from xhtml1 to html5, although from what I can see on [this page](http://methods.co.nz/asciidoc/userguide.html#X5), the html5 back-end is 'functionally identical' to the xhtml1 one, so I don't know how much of an upgrade that would be. I looked further and bumped into [asciidoctor](https://asciidoctor.org/), which seems more modern and is also what Github uses to support AsciiDoc format. Are you ok with me looking into this a bit further and finding a way to output cleaner HTML for the docs?

Yes, but in a separate PR please. I’m using asciidoctor for https://robustirc.net/docs, and it is indeed nicer.

> For backwards compatibility we could keep the xhtml1.css file and keep it linked to all the docs that are already built, and link to the new style.css file for any newer docs.

Sounds good.

> Edit: Are you familiar with [this project](https://github.com/asciidoctor/jekyll-asciidoc)? Seems very interesting.

No, but note that it’s most likely not available in the jekyll that GitHub pages uses, so it’s not going to be of much use to us, I think.

**Comment by dreadnip** [comment]
I've added commits to update each major sub-section of the site except for the screenshots/gallery page. What is your opinion on removing the JS gallery entirely? While it still works, it doesn't scale well on mobile and has no swipe interaction. It has a couple of old dependencies that we could drop and it doesn't add that much value to the site from my point of view.

Docs are still broken so this PR can be regarded as WIP.

**Comment by stapelberg** [comment]
> What is your opinion on removing the JS gallery entirely? While it still works, it doesn't scale well on mobile and has no swipe interaction. It has a couple of old dependencies that we could drop and it doesn't add that much value to the site from my point of view.

I’d rather leave it on there until it actually breaks in hard-to-fix ways. It’s better to have a convenient gallery on non-mobile than to have none at all.

**Comment by stapelberg** [comment]
What’s the status of these changes?

They need to be updated to address merge conflicts at the very least, but is the rest ready to review? There was no update after my most recent comment AFAICT.

**Comment by dreadnip** [comment]
I was working on the gallery and videos pages but completely forgot about it. There wasn't any interaction in this thread so I also figured you guys didn't really care for it. I'll take a look at it this afternoon and see if I can wrap up the PR excluding the docs conversion :)

**Comment by dreadnip** [comment]
I've fixed the merge conflicts and finished converting the screenshot gallery code to vanilla JS. Let me know what you think. I had to rebuild the docs because the header HTML structure changed slightly. I know we said to keep that in a separate PR, but I had to include just the header change in here to not break the site once merged.

I'll do some more work on making the docs HTML5 compatible and fully responsive if you decide to pick up this PR. I can also re-write the table of contents and footnotes JS (no fucntional changes, just refactoring the JS into this century).

File: index.html [hunk 3]
```diff
@@ -3,111 +3,106 @@
...
-        <a href="/downloads">
-        <span style="font-weight: bold; color: #3A8ECD; margin-right: .5em">➡</span>
-        Download the latest version
-        <span style="margin-left: 2em; color: #c0c0c0">4.19</span>
-        </a>
-    </div>
-</div>
+    <a class="screen" href="screenshots/" title="view more screenshots of i3">
+        <img src="screenshots/i3-14.bigthumb.png" alt="i3 screenshot"
+             srcset="screenshots/i3-14.bigthumb.jpg 2x">
```
Comment (stapelberg) [inline index.html:23]: i3-14.bigthumb.jpg doesn’t exist, but i3-14.bigthumb.png does (png instead of jpg).

File: index.html [hunk 4]
```diff
@@ -3,111 +3,106 @@
...
-<div id="info" class="clearfix">
+<section class="info">
 
-    <div id="screens">
-        <a href="/screenshots/" title="view more screenshots of i3">
-        <img id="screenshot" src="/screenshots/i3-9.bigthumb.png" width="375" height="234" alt="i3 screenshot"
-	 srcset="/screenshots/i3-12.bigthumb.jpg 2x">
+    <div class="info-text">
+        <h2>Do What I Mean. Good Docs. Clean Code. Sounds good?</h2>
+        <p>
```
Comment (stapelberg) [inline index.html:10]: There’s almost no space between heading, paragraph and download button. I think the stylesheet needs explicit `margin` definitions for basic elements such as headings and paragraphs.

File: downloads/index.html [hunk 5]
```diff
@@ -1,62 +1,50 @@
...
-  The current stable version is 4.19.
-</p>
+<p>The current stable version is 4.19.</p>
 
-<p>
-All releases are crytographically signed with the following <a href="https://gnupg.org/">GnuPG</a> key:
-<pre>
+<p>All releases are crytographically signed with the following <a href="https://gnupg.org/">GnuPG</a> key:
+
+<blockquote class="gpg-key">
```
Comment (stapelberg) [inline downloads/index.html:37]: This blockquote shows an unnecessary horizontal scrollbar in my browser. Could you remove that please?

File: docs/index.html [hunk 6]
```diff
@@ -3,142 +3,101 @@
...
+<h2>Documentation for i3 v4.19</h2>
 
-<br style="clear: both">
+<p>One of i3’s goals is good documentation. The documents which you will find below will hopefully answer all your questions. If you have any corrections or suggestions please let us know!</p>
+
+<section class="columns">
+    <div class="docs">
+        <h2>For users</h2>
+        <div class="item highlight">
+            <a href="docs/userguide.html">User’s Guide</a>
```
Comment (stapelberg) [inline docs/index.html:15]: When using `jekyll serve` and accessing http://localhost:4000/docs/, the userguide link no longer works: it leads me to http://localhost:4000/docs/docs/userguide.html (note the extra `docs` in the URL).

Should we stick with absolute links for now?

File: index.html [hunk 7]
```diff
@@ -3,111 +3,106 @@
...
-values when considering contributions to the project:
+    i3 is a <a href="http://en.wikipedia.org/wiki/Tiling_window_manager">tiling
+    window manager</a>, completely written from scratch. The target platforms are
+    GNU/Linux and BSD operating systems, our code is Free and Open Source Software
+    (FOSS) under the BSD license. i3 is primarily targeted at advanced users and
+    developers. Based upon the experiences we made when wanting to hack/fix wmii,
+    we agreed upon the following goals for i3:
 </p>
 
 <ul>
```
Comment (stapelberg) [inline index.html:37]: Can you leave this as an `<ol>` for now please? These are actually ordered.

We can revise when we change the content as discussed in the other issue.

File: _layouts/default.html [hunk 8]
```diff
@@ -1,79 +1,39 @@
...
+        </header>
+        <main>
+	        {{ content }}
+        </main>
+        <footer>
+            © 2009-present Michael Stapelberg,
+            <a href="/impress.html">Impressum</a>,
+            <a href="https://github.com/i3/i3.github.io">Source</a>
+        </footer>
+        <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.5.1/jquery.min.js"></script>
```
Comment (stapelberg) [inline _layouts/default.html:37]: Can you make this inclusion of jquery (if it’s still required?) conditional on there being any javascript on the page please? It currently shows up on the index page, for example.

**Review by stapelberg — COMMENTED** [review]
Thanks very much for picking this back up again, and sorry for the slow response.

I like the new gallery, and aside from a few issues I flagged, I think we can go forward with this PR.

File: index.html (hunk 7 above)
Comment (dreadnip) [reply index.html]: Didn't actually see it was an ol, fixed.

File: _layouts/default.html (hunk 8 above)
Comment (dreadnip) [reply _layouts/default.html]: 62335369104fd0c7681971aa496b74fa1b238b8f

File: index.html (hunk 4 above)
Comment (dreadnip) [reply index.html:10]: a7983848b6c3c1a7d7eb151b9cd3da259e2ffeac

File: downloads/index.html (hunk 5 above)
Comment (dreadnip) [reply downloads/index.html:37]: 11a9af9a5005fdebea71af55d4e27b000e30c905

File: index.html (hunk 3 above)
Comment (dreadnip) [reply index.html]: 8bcf0293deed62850314550287eb57d251f446dd

File: docs/index.html (hunk 6 above)
Comment (dreadnip) [reply docs/index.html]: be448f84f45028e82c6453ca7ee1c114137d85aa

File: css/style.css [hunk 9]
```diff
@@ -1,152 +1,116 @@
...
+nav ul li {
+    list-style: none;
+}
+
+nav ul li a {
     color: #FFF;
+    padding: .4rem .8rem;
     text-decoration: none;
     border-bottom: 2px solid #2e2e2e;
+    transition: ease 150ms;
```
Comment (stapelberg) [inline css/style.css:79]: Can you remove all animations/transitions please?

i3 itself doesn’t have any animations/transitions itself, so it seems weird to add some here.

File: index.html [hunk 10]
```diff
@@ -3,111 +3,106 @@
...
-    <div id="download">
-        <a href="/downloads">
-        <span style="font-weight: bold; color: #3A8ECD; margin-right: .5em">➡</span>
-        Download the latest version
-        <span style="margin-left: 2em; color: #c0c0c0">4.19</span>
-        </a>
-    </div>
-</div>
+    <a class="screen" href="screenshots/" title="view more screenshots of i3">
+        <img src="screenshots/i3-14.bigthumb.png" alt="i3 screenshot"
```
Comment (stapelberg) [inline index.html:22]: Can you stick with the current screenshot please? The one visible right now does not use the default config (on the contrary, it visibly deviates from the default), and I don’t want to give people the wrong impression about what they’re going to get once they install i3.

File: docs/index.html [hunk 11]
```diff
@@ -3,142 +3,101 @@
...
+
+<section class="columns">
+    <div class="docs">
+        <h2>For users</h2>
+        <div class="item highlight">
+            <a href="/docs/userguide.html">User’s Guide</a>
+            <p>Introduction and reference. Read this!</p>
+        </div>
+        <div class="item">
+            <a href="conduct">Code of Conduct</a>
```
Comment (stapelberg) [inline docs/index.html:19]: This link still isn’t absolute.

Could you run a link checker on your working copy to make sure we don’t accidentally introduce any broken links? As you can see, it’s easy to miss them :)

File: css/style.css (hunk 9 above)
Comment (dreadnip) [reply css/style.css]: 6f38a84

File: index.html (hunk 10 above)
Comment (dreadnip) [reply index.html]: 242b610

File: docs/index.html (hunk 11 above)
Comment (dreadnip) [reply docs/index.html]: See a89ade7

I ran a linkchecker and only got two 404s. https://i3wm.org/downloads/RELEASE-NOTES-3.a-bf2.txt and https://i3wm.org/downloads/RELEASE-NOTES-3.a.txt, both of which also 404 on the live site. I also got some hits on SVGs on the refcard, but I can look to update the refcard in a separate PR. It seems to render identical to the live version.

**Review by stapelberg — APPROVED** [review]

**Comment by stapelberg** [comment]
Looks good! Could you rebase your changes please?

I gave it a quick try, but it’s a lot of changes and I don’t want to introduce any mistakes.

**Comment by stapelberg** [comment]
Hmm, GitHub still says:

> This branch cannot be rebased due to conflicts
> Rebasing the commits of this branch on top of the base branch cannot be performed automatically due to conflicts encountered while reapplying the individual commits from the head branch.

You only merged, but didn’t rebase, yes?

**Comment by dreadnip** [comment]
I ran `git rebase master` on my feature branch locally. I fixed the conflicts that appeared between my changes and the new commits on master. Then I force pushed this branch so the PR would update. Github still indicated conflicts so I fixed those manually using the Github interface.

**Comment by stapelberg** [comment]
Thanks for the details. I guess I’ll just go with a normal merge instead of a rebase merge, then.

It’ll be less easy to navigate in git (which is why I normally only do rebase merges), but at least your individual commits will be retained.
